File: dart/2313.dart
import 'dart:io';

void main() {
  List<String> l = stdin.readLineSync()!.split(" ");
  int a = int.parse(l[0]), b = int.parse(l[1]), c = int.parse(l[2]);

  if (a + b > c && b + c > a && a + c > b) {
    String output = "";
    String r = "N";
    if (a == b && b == c) {
      output = "Equilatero";
    } else {
      if (a * a + b * b == c * c ||
          b * b + c * c == a * a ||
          a * a + c * c == b * b) {
        r = "S";
      }
      if (a == b || b == c || a == c) {
        output = "Isoceles";
      } else {
        output = "Escaleno";
      }
    }

    print("Valido-${output}\nRetangulo: ${r}");
  } else {
    print("Invalido");
  }
}

File: python/2313.py
a, b, c = map(int, input().split())
t = ""
r = "N"

if a+b > c and b+c > a and a+c > b:
    if a == b and b == c:
        t = "Equilatero"
    else:
        if a**2 + b**2 == c**2 or b**2 + c**2 == a**2 or a**2 + c**2 == b**2:
            r = "S"
        if a == b or b == c or c == a:
            t = "Isoceles"
        else:
            t = "Escaleno"

    print("Valido-%s\nRetangulo: %s" %(t, r))
else:
    print("Invalido")
